langchain/src/chat_models/azure_ml.ts

```diff
@@ -0,0 +1,142 @@
+}
+
+export class LlamaContentFormatter implements ChatContentFormatter {
+  _convertMessageToRecord(message: BaseMessage): Record<string, unknown> {
+    if (message._getType() === "human") {
+      return { role: "user", content: message.content };
+    } else if (message._getType() === "ai") {
+      return { role: "assistant", content: message.content };
+    } else {
+      return { role: message._getType(), content: message.content };
+    }
+  }
+
+  formatRequestPayload(
+    messages: BaseMessage[],
+    modelArgs: Record<string, unknown>
+  ): string {
+    let msgs = messages.map((message) => {
+      this._convertMessageToRecord(message);
+    });
+    return JSON.stringify({
+      input_data: {
+        input_string: msgs,
+        parameters: modelArgs,
+      },
+    });
+  }
+
+  formatResponsePayload(responsePayload: string) {
+    const response = JSON.parse(responsePayload);
+    return response.output;
+  }
+}
+
+/**
+ * Type definition for the input parameters of the AzureMLChatOnlineEndpoint class.
+ */
+export interface AzureMLChatParams extends BaseChatModelParams {
+  endpointUrl?: string;
+  endpointApiKey?: string;
+  modelArgs?: Record<string, unknown>;
+  contentFormatter?: ChatContentFormatter;
+}
+
+/**
+ * Class that represents the chat model. It extends the SimpleChatModel class and implements the AzureMLChatInput interface.
+ */
+export class AzureMLChatOnlineEndpoint
+  extends SimpleChatModel
+  implements AzureMLChatParams
+{
+  static lc_name() {
+    return "AzureMLChatOnlineEndpoint";
+  }
+  static lc_description() {
+    return "A class for interacting with AzureML Chat models.";
+  }
+  endpointUrl: string;
+  endpointApiKey: string;
+  modelArgs?: Record<string, unknown>;
+  contentFormatter: ChatContentFormatter;
+  httpClient: AzureMLHttpClient;
+
+  constructor(fields: AzureMLChatParams) {
+    super(fields ?? {});
+    if (!fields?.endpointUrl && !getEnvironmentVariable("AZUREML_URL")) {
+      throw new Error("No Azure ML Url found.");
+    }
+    if (!fields?.endpointApiKey && !getEnvironmentVariable("AZUREML_API_KEY")) {
+      throw new Error("No Azure ML ApiKey found.");
+    }
+    if (!fields?.contentFormatter) {
+      throw new Error("No Content Formatter provided.");
+    }
+
+    this.endpointUrl =
+      fields.endpointUrl || getEnvironmentVariable("AZUREML_URL") + "";
+    this.endpointApiKey =
+      fields.endpointApiKey || getEnvironmentVariable("AZUREML_API_KEY") + "";
+    this.httpClient = new AzureMLHttpClient(
+      this.endpointUrl,
+      this.endpointApiKey
+    );
+    this.contentFormatter = fields.contentFormatter;
+    this.modelArgs = fields?.modelArgs;
+  }
+  get _identifying_params() {
+    const modelKwargs = this.modelArgs || {};
+    return {
+      ...super._identifyingParams,
+      modelKwargs,
+    };
+  }
+
+  _llmType() {
+    return "azureml_chat";
+  }
+
+  _combineLLMOutput(): Record<string, any> | undefined {
+    return [];
```

Comment:
If it's not implemented, it should throw a not implemented error.
```suggestion
    throw new Error("AzureMLChatOnlineEndpoint._combineLLMOutput called, but is not implemented.")
```